app/Alexa/Models/User.php

```diff
@@ -86,13 +86,67 @@ public function hasTeam()
         return $this->team_id != null;
     }
 
+    /**
+     * Check if this user has an eligible team.
+     *
+     * @return bool
+     */
+    public function hasEligibleTeam()
+    {
+        return $this->hasTeam() && $this->team->isEligibleForAcceptance();
+    }
+
     /**
      * Check if this user completed the application.
      *
      * @return bool
      */
     public function hasCompleteApplication()
     {
-        return $this->hasCompleteProfile() && $this->hasTeam();
+        return $this->hasCompleteProfile() && $this->hasEligibleTeam();
+    }
+
+    /**
+     * Scope a query to only include users that match given search.
+     *
+     * @param  $query  \Illuminate\Database\Eloquent\Builder
+     * @param  string|null  $search
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeFilterByEmailOrNickname($query, $search)
+    {
+        if ($search === null) {
+            return $query;
+        }
+
+        return $query->where('email', 'ILIKE', $search.'%')
```

Comment:
The `ILIKE` query function is specific from `PostgresSQL`. I think we should or update the `.env.example`, or document this requirement, or add an option to others database engines.

Reply:
Done 👍